src/libs/actions/Session/index.js

```diff
@@ -23,6 +23,32 @@ Onyx.connect({
     callback: val => credentials = val || {},
 });
 
+/**
+ * Manage push notification subscriptions on sign-in/sign-out.
+ *
+ * We would normally do this when AuthScreens mounts and unmounts. However on Android,
+ * AuthScreens unmounts when the app is closed with the back button so we manage the
+ * push subscription when the session changes here.
+ */
+let previousAccountID = '';
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (session) => {
+        const accountID = lodashGet(session, 'accountID');
+
+        if (accountID && previousAccountID !== accountID) {
```

Comment:
So if we have an `accountID` and the `previousAccountID` doesn't equal it then we register a new account.

Are we assuming the `previousAccountID === ''` here or do we not care about what value it was before?

Reply:
I guess I'm wondering if there is a case where the `accountID` can have a value but be different from `previousAccountID`. I'm not sure if that's possible though - maybe it always has to be at least `null`.

Reply:
Yeah as long as the value changed, we would want to register the new account ID. This should replace the old registration if there was one.